net/tailscale/files/tailscale.init

```diff
@@ -4,44 +4,263 @@
+      else
+        ts_ip=$(tailscale ip -4)
+        uci set network.tailscale.proto='static'
+        uci set network.tailscale.ipaddr=$ts_ip
+        uci set network.tailscale.netmask='255.0.0.0'
+        uci set network.tailscale.device=$ts0
+      fi
+    fi
+
+    # Find the LAN to WAN forwarding rule index
+    index=$(uci show firewall | grep "firewall.@forwarding\[[0-9]\+\]\.src='lan'" -B 1 -A 1 | grep "firewall.@forwarding\[[0-9]\+\]\.dest='wan'" | grep -o '[0-9]\+')
+    # Configure firewall forwarding rules based on exit_node
+    if [ -n "$exit_node" ]; then
+      uci set firewall.@defaults[0].forward='REJECT'
+      [ -n $index ] && uci set firewall.@forwarding[$index].enabled='0'
+    fi
+
+    # Configure subnet routes for site to site
+    config_get subnet_routes $cfg 'subnet_routes'
+    if [ -n "$subnet_routes" ]; then
+      i=1
+      ts_ip=$(tailscale ip -4)
+      for route in $subnet_routes; do
+        uci set network.ts_subnet$i='route'
+        uci set network.ts_subnet$i.interface='tailscale'
+        uci set network.ts_subnet$i.target=$route
+        uci set network.ts_subnet$i.gateway=$ts_ip
+        let i++
+      done
+    fi
+
+    # Create firewall zone and forwarding rules for Tailscale
+    config_get access $cfg 'access'
+    if [ -n "$access" ]; then
+      uci batch <<-EOF
+        set firewall.tszone='zone'
+        set firewall.tszone.input='ACCEPT'
+        set firewall.tszone.output='ACCEPT'
+        set firewall.tszone.forward='ACCEPT'
+        set firewall.tszone.masq='1'
+        set firewall.tszone.mtu_fix='1'
+        set firewall.tszone.name='tailscale'
+        set firewall.tszone.network='tailscale'
+EOF
+    fi
+    # Configure specific firewall forwarding rules between Tailscale, LAN, and WAN.
+    if [ "${access//ts_ac_lan/}" != "$access" ]; then
+      uci set firewall.ts_ac_lan=forwarding
+      uci set firewall.ts_ac_lan.dest='lan'
+      uci set firewall.ts_ac_lan.src='tailscale'
+    fi
+    if [ "${access//ts_ac_wan/}" != "$access" ]; then
+      uci set firewall.ts_ac_wan=forwarding
+      uci set firewall.ts_ac_wan.dest='wan'
+      uci set firewall.ts_ac_wan.src='tailscale'
+    fi
+    if [ "${access//lan_ac_ts/}" != "$access" ]; then
+      uci set firewall.lan_ac_ts=forwarding
+      uci set firewall.lan_ac_ts.dest='tailscale'
+      uci set firewall.lan_ac_ts.src='lan'
+    fi
+    if [ "${access//wan_ac_ts/}" != "$access" ]; then
+      uci set firewall.wan_ac_ts=forwarding
+      uci set firewall.wan_ac_ts.dest='tailscale'
+      uci set firewall.wan_ac_ts.src='wan'
+    fi
+
+    # Commit configuration changes and reload service
+    [ -n "$(uci changes dhcp)" ] && uci commit dhcp && /etc/init.d/dnsmasq reload
+    [ -n "$(uci changes network)" ] && uci commit network && /etc/init.d/network reload
+    [ -n "$(uci changes firewall)" ] && uci commit firewall && /etc/init.d/firewall reload
+
+    # Delete lock file
+    trap 'rm -f "$LOCK_FILE"; exit' INT TERM EXIT
```

Comment:
You should traps go _before_ the code that might fail, so they are always defined.